source/process/documentation-UItext-guidelines.rst

```diff
@@ -0,0 +1,193 @@
+- Include a question that has a Yes/No answer.
+- Examples:
+
+  - *Are you sure you want to delete this channel?*
+  - *A plug-in with this ID already exists. Would you like to overwrite it?*
+
+Warning messages
+~~~~~~~~~~~~~~~~
+
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- If the message contains a question, phrase it in such a manner so that it has a Yes/No answer (unless you have specific action buttons for the message).
+
+- Examples:
+
+  - *The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.*
+  - *If you claim this AD/LDAP account, you will no longer be able to log in with your email. Do you want to continue?*
+
+Error messages
+~~~~~~~~~~~~~~~
+
+An error message is for telling somebody that something went wrong. Errors prevent a user from doing a task or accessing a feature till the error is resolved.
+
+- Use complete sentences.
+- If what went wrong isn't obvious, explain in one sentence.
+- If a solution or workaround isn't obvious, suggest one.
+- Examples:
+
+  - *This message is too long. Shorten it to 120 characters.*
+  - *The passwords do not match.*
+
+
+Guidelines for UI elements
+---------------------------
+Use this table when writing the text for UI elements such as windows, dialog boxes, labels, and prompts.
+
+.. list-table::
+   :widths: 10 30 10 15 35
+   :header-rows: 1
+
+   * - Element
+     - Image 
+     - Capitalization
+     - Phrasing
+     - Examples     
+   * - Menu
+     - |menu.png|
+     - All words
+     - - Noun, noun phrase, or verb
+       - No punctuation
+       - Not more than three words
+     - - *Members*
+       - *Account Preferences*
+       - *Log Out*
+   * - Tooltip
+     - |tooltip.png|
+     - First word
+     - - Sentence fragment or sentence
+       - No punctuation
+       - Include articles (*a*, *an*, *the*)
+     - - *Start a Zoom meeting*
+       - *Flag for following up*
+       - *Remove from this list*
+   * - In-field text
+     - |field.png|  
+     - First word
+     - - Sentence fragment, sentence, or word
+       - No punctuation
+       - Include articles (*a*, *an*, *the*)
+     - - *Add a comment*
+       - *Search*
+   * - Action button
+     - |action.png|
+     - All words
+     - - Verb or verb phrase
+       - No articles (*a*, *an*, *the*)
+       - Exceptions: *OK*, *Yes*, *No*
+     - - *Add Comment*
+       - *Edit*
+   * - Label before a UI element
+     - |label_before.png|
+     - First word
+     - - Noun, verb, or sentence fragment
+       - End with a colon
+       - Include articles (*a*, *an*, *the*)
+     - - *Sign in with:*
+       - *Other words, separated by commas:*
+   * - Label after a UI element
+     - |label_after.png|
```

Comment:
Notes for this category:

1. We need to find consistency in the product. For example, do we refer to things as "Off" or "Never"
2. How much description is used for these. ie do we rely more on help text to keep the radio buttons easier to read
3. Max length guidelines required?

Reply:
1. We could probably have a word-usage list to lay down which words should be used (or not) when.
2. My preference would be radio buttons because...self-contained.
3. Not sure. Keep it open, and edit the edge cases (if any) during PR review?

Reply:
1. Agree, and ideally update the guidelines or link to that for reference if applicable 0/5
2. We can iterate, but one thing to keep in mind is if radio button text gets long it can be overwhelming for a user to read everything and change settings vs trying to make the radio buttons as concise as possible and explain only what's needed to fill in the gaps in help text.
3. That's fine, again I would propose a preference of conciseness

Reply:
Added "Brief (move lengthier explanations belong to help text)"

For word-usage list, okay if I add links to the [MSTP list](https://docs.microsoft.com/en-us/style-guide/welcome/) (we need to click on the A-Z word collection link on the ToC at the left; doesn't have a direct URL) in the general guidelines section?